### PR TITLE
Make the testing process on different platforms more consistent

### PR DESCRIPTION
**Pull request by xster**
Doing it in 3 steps. Going to change LUCI before cleaning up more.

### CODE REVIEW

File: testing/scenario_app/ios/Scenarios/Scenarios.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -451,27 +450,6 @@
 		};
 /* End PBXResourcesBuildPhase section */
 
-/* Begin PBXShellScriptBuildPhase section */
-		0D8470A8241035AD0030B565 /* Compile iOS JIT */ = {
```
Comment (xster) [inline testing/scenario_app/ios/Scenarios/Scenarios.xcodeproj/project.pbxproj:455]: it didn't make sense to me that we would auto build the dart code (which isn't even visible in the xcode project) when running xcode but not build the engine

File: testing/run_tests.py [hunk 2]
```diff
@@ -282,13 +284,47 @@ def EnsureJavaTestsAreBuilt(android_out_dir):
   RunCmd(gn_command, cwd=buildroot_dir)
   RunCmd(ninja_command, cwd=buildroot_dir)
 
+def EnsureIosTestsAreBuilt(ios_out_dir):
```
Comment (gaaclarke) [inline testing/run_tests.py:287]: I know we aren't doing this in this file but a small docstring would be helpful here.  This file is growing and getting harder to maintain.  (I'm sure you know but you put docstrings inside python functions)

File: testing/run_tests.py [hunk 3]
```diff
@@ -282,13 +284,47 @@ def EnsureJavaTestsAreBuilt(android_out_dir):
...
   EXPECTED_VERSION = '1.8'
   # `java -version` is output to stderr. https://bugs.java.com/bugdatabase/view_bug.do?bug_id=4380614
   version_output = subprocess.check_output(['java', '-version'], stderr=subprocess.STDOUT)
   match = bool(re.compile('version "%s' % EXPECTED_VERSION).search(version_output))
   message = "JUnit tests need to be run with Java %s. Check the `java -version` on your PATH." % EXPECTED_VERSION
-  assert True, message
+  assert match, message
+
+def AssertExpectedXcodeVersion():
+  EXPECTED_MAJOR_VERSION = '11'
```
Comment (gaaclarke) [inline testing/run_tests.py:324]: This is probably better suited as an arg to this function, no?

Nit: return a bool and assert at a higher level

**Comment by xster** [comment]
It will until I merge https://flutter-review.googlesource.com/c/recipes/+/3160 after this one. Then I'll clean up more.

File: testing/run_tests.py (hunk 2 above)
Comment (xster) [reply testing/run_tests.py:288]: ya for sure. Added

File: testing/run_tests.py (hunk 3 above)
Comment (xster) [reply testing/run_tests.py:328]: This file's just a rather ad-hoc script. I doubt we'll ever touch this line again.

**Pull request merged**